core/src/gl/renderState.cpp

```diff
@@ -21,4 +21,12 @@ namespace RenderState {
         glClearColor(0.3, 0.3, 0.3, 1.0);
     }
     
+    void invalidateAllStates() {
+        blending.invalidate();
+        culling.invalidate();
+        depthTest.invalidate();
+        depthWrite.invalidate();
+        blendingFunc.invalidate();
+    }
+    
```

Comment:
We get a call to `onContextDestroyed()` whenever invalidation is needed, so an extra state variable seems unnecessary